prql-compiler/src/parser/mod.rs

```diff
@@ -1022,23 +1104,19 @@ Canada
               Integer: 2
         "###);
 
-        // Underscores can't go at the beginning or end of numbers
-        assert!(expr_of_string("_2", Rule::number).is_err());
-        assert!(expr_of_string("_", Rule::number).is_err());
-        assert!(expr_of_string("_2.3", Rule::number).is_err());
-        // We need to test these with `stmts_of_string` because they start with
-        // a valid number (and pest will return as much as possible and then return)
-        let bad_numbers = vec!["2_", "2.3_"];
-        for bad_number in bad_numbers {
-            assert!(stmts_of_string(bad_number).is_err());
-        }
-        Ok(())
+        // Underscores at the beginning are parsed as ident
+        expr_of_string("_2").unwrap().kind.into_ident().unwrap();
+        expr_of_string("_").unwrap().kind.into_ident().unwrap();
+
+        expr_of_string("_2.3").unwrap_err();
+        // expr_of_string("2_").unwrap_err(); // TODO
```

Comment:
All OK with these? They actually took a while in pest, happy to help with them in chumsky...

Reply:
Ah yeah, this is also todo